cdap-ui/app/cdap/components/CaskWizards/MicroserviceUpload/MicroserviceQueue.js

```diff
@@ -0,0 +1,203 @@
+              onChange={this.props.onChange.bind(null, 'namespace')}
+            />
+          </Col>
+        </FormGroup>
+        {this.renderFormGroupProperty('topic', true)}
+      </div>
+    );
+  }
+
+  renderSQSProperties() {
+    if (this.props.type !== 'sqs') { return null; }
+
+    return (
+      <div>
+        {this.renderFormGroupProperty('region', true)}
+        {this.renderFormGroupProperty('accessKey', true, 'password')}
+        {this.renderFormGroupProperty('accessId', true, 'password')}
+        {this.renderFormGroupProperty('queueName', true)}
+        {this.renderFormGroupProperty('endpoint')}
+      </div>
+    );
+  }
+
+  renderWebsocketsProperties() {
+    if (this.props.type !== 'websockets') { return null; }
+
+    return (
+      <div>
+        {this.renderFormGroupProperty('connection', true)}
+        {this.renderFormGroupProperty('sslKeystoreFilePath')}
+        {this.renderFormGroupProperty('sslKeystorePassword', false, 'password')}
+        {this.renderFormGroupProperty('sslKeystoreKeyPassword', false, 'password')}
+        {this.renderFormGroupProperty('sslKeystoreType')}
+        {this.renderFormGroupProperty('sslTruststoreFilePath')}
+        {this.renderFormGroupProperty('sslTruststorePassword', false, 'password')}
+        {this.renderFormGroupProperty('sslTruststoreType')}
+      </div>
+    );
+  }
+
+  render() {
+    return (
+      <div className="microservice-queue-container">
+        <div className="col-xs-10 microservice-queue-input-container">
+          <Form
+            className="form-horizontal"
+            onSubmit={(e) => {
+              e.preventDefault();
+              return false;
+            }}
+          >
+            <FormGroup row>
+              <Col xs="3">
+                <Label className="control-label">{T.translate('commons.nameLabel')}</Label>
+              </Col>
+              <Col xs="8">
+                <Input
+                  type="text"
+                  value={this.props.name}
+                  autoFocus
+                  onChange={this.props.onChange.bind(null, 'name')}
+                  className="form-control"
+                />
+              </Col>
+              <IconSVG
+                name="icon-asterisk"
+                className="text-danger"
+              />
+            </FormGroup>
+            <FormGroup row>
+              <Col xs="3">
+                <Label className="control-label">{T.translate('commons.typeLabel')}</Label>
+              </Col>
+              <Col xs="8">
+                <SelectWithOptions
+                  options={defaultQueueTypes}
+                  value={this.props.type}
+                  onChange={this.props.onChange.bind(null, 'type')}
+                />
+              </Col>
+              <IconSVG
+                name="icon-asterisk"
+                className="text-danger"
+              />
+            </FormGroup>
+            {this.renderTMSProperties()}
```

Comment:
@ajainarayanan I think @nitinmotgi had a good point though, so instead of calling all three `renderTMSProperties()`, `renderSQSProperties()` etc. in the `render()` fn, now I just call `renderTypeProperties()`, which takes care of calling the right render functions depending on the type. Same functionality, but it's a bit cleaner this way I think.